MatrixSDK/Crypto/Dehydration/MXDehydrationService.m

```diff
@@ -0,0 +1,295 @@
+// 
+// Copyright 2021 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "MXDehydrationService.h"
+#import <OLMKit/OLMKit.h>
+#import "MXCrypto.h"
+#import "MXCrypto_private.h"
+#import "MXCryptoTools.h"
+#import "MXKeyProvider.h"
+#import "MXCrossSigning_Private.h"
+#import "MXRawDataKey.h"
+#import "MXSession.h"
+
+NSString *const MXDehydrationAlgorithm = @"org.matrix.msc2697.v1.olm.libolm_pickle";
+
+NSString *const MXDehydrationServiceKeyDataType = @"org.matrix.sdk.dehydration.service.key";
+
+@interface MXDehydrationService()
+{
+    BOOL inProgress;
```

Comment:
Could we expose it publicly?

That will avoid to call `dehydrateDeviceWithMatrixRestClient` when it is useless.